tests/actions.go

```diff
@@ -792,157 +817,29 @@ func (oa *operatorActions) UpgradeTidbClusterOrDie(info *TidbClusterConfig) {
-			}
-			if !imageUpgraded(v1alpha1.TiKVMemberType, tikvSet) {
-				return false, pingcapErrors.New("tikv image is not updated while tikv is upgrading")
-			}
-			if !setUpgraded(tikvSet) {
-				if imageUpgraded(v1alpha1.TiDBMemberType, tidbSet) {
-					return false, pingcapErrors.New("tidb image is updated while tikv is upgrading")
-				}
-			}
-			return false, nil
-		} else if tc.Status.TiDB.Phase == v1alpha1.UpgradePhase {
-			glog.Infof("tidb is upgrading")
-			if !imageUpgraded(v1alpha1.PDMemberType, pdSet) {
-				return false, pingcapErrors.New("pd image is not updated while tidb is upgrading")
-			}
-			if !setUpgraded(pdSet) {
-				return false, pingcapErrors.New("pd stateful set is not upgraded while tidb is upgrading")
-			}
-			if !imageUpgraded(v1alpha1.TiKVMemberType, tikvSet) {
-				return false, pingcapErrors.New("tikv image is not updated while tidb is upgrading")
-			}
-			if !setUpgraded(tikvSet) {
-				return false, pingcapErrors.New("tikv stateful set is not upgraded while tidb is upgrading")
-			}
-			if !imageUpgraded(v1alpha1.TiDBMemberType, tidbSet) {
-				return false, pingcapErrors.New("tidb image is not updated while tikv is upgrading")
-			}
-			return false, nil
-		}
-
-		// check pd final state
-		if !imageUpgraded(v1alpha1.PDMemberType, pdSet) {
-			return false, nil
-		}
-		if !setUpgraded(pdSet) {
-			glog.Infof("check pd stateful set upgraded failed")
-			return false, nil
-		}
-		// check tikv final state
-		if !imageUpgraded(v1alpha1.TiKVMemberType, tikvSet) {
-			return false, nil
-		}
-		if !setUpgraded(tikvSet) {
-			glog.Infof("check tikv stateful set upgraded failed")
-			return false, nil
-		}
-		// check tidb final state
-		if !imageUpgraded(v1alpha1.TiDBMemberType, tidbSet) {
-			return false, nil
-		}
-		if !setUpgraded(tidbSet) {
-			glog.Infof("check tidb stateful set upgraded failed")
-			return false, nil
-		}
-		return true, nil
-	})
-}
-
 func (oa *operatorActions) DeployMonitor(info *TidbClusterConfig) error { return nil }
 func (oa *operatorActions) CleanMonitor(info *TidbClusterConfig) error  { return nil }
 
-func getComponentContainer(set *v1beta1.StatefulSet) (corev1.Container, bool) {
-	name := set.Labels[label.ComponentLabelKey]
-	for _, c := range set.Spec.Template.Spec.Containers {
-		if c.Name == name {
-			return c, true
+func getPodContainer(kubeCli kubernetes.Interface, namespace string, memberName string) (*corev1.Container, bool) {
+	name := fmt.Sprintf("%s-%d", memberName, 0)
+	pod, err := kubeCli.CoreV1().Pods(namespace).Get(name, metav1.GetOptions{})
+	if err != nil {
+		glog.Errorf("fail to get pod [%s/%s]", namespace, name)
+		return nil, false
+	}
+	if len(pod.Spec.Containers) == 0 {
+		glog.Errorf("no container in this pod [%s/%s]", namespace, name)
+		return nil, false
+	}
+
+	for _, container := range pod.Spec.Containers {
+		if container.Name == label.TiDBLabelVal ||
+			container.Name == label.TiKVLabelVal ||
+			container.Name == label.PDLabelVal {
```

Comment:
```suggestion
		if container.Name == v1alpha1.PDMemberType.String() ||
```
Though the literal value is same coincidentally, we should use `v1alpha1.XXMemberType` here (So do `TiDB` and `TiKV`).

Reply:
done